app/api/custom_placeholders.py

```diff
@@ -0,0 +1,148 @@
+                                   related_view_kwargs={'custom_placeholder_id': '<id>'},
+                                   schema='EventSubTopicSchema',
+                                   type_='event-sub-topic')
+
+
+class CustomPlaceholderList(ResourceList):
+    """
+    List and create event custom placeholders
+    """
+
+    def query(self, view_kwargs):
+        """
+        query method for custom placeholders list
+        :param view_kwargs:
+        :return:
+        """
+        query_ = self.session.query(CustomPlaceholder)
+        if view_kwargs.get('event_sub_topic_id'):
+            event_sub_topic = safe_query(self, EventSubTopic, 'id', view_kwargs['event_sub_topic_id'],
+                                         'event_sub_topic_id')
+            query_ = query_.join(EventSubTopic).filter(EventSubTopic.id == event_sub_topic.id)
+        return query_
+
+    def before_create_object(self, data, view_kwargs):
+        """
+        method to create object before post
+        :param data:
+        :param view_kwargs:
+        :return:
+        """
+        if view_kwargs.get('event_sub_topic_id'):
+            event_sub_topic = safe_query(self, EventSubTopic, 'id', view_kwargs['event_sub_topic_id'],
+                                         'event_sub_topic_id')
+            data['event_sub_topic_id'] = event_sub_topic.id
+
+    def after_create_object(self, custom_placeholder, data, view_kwargs):
+        if data.get('original_image_url'):
+            uploaded_images = create_save_image_sizes(data['original_image_url'], 'custom-placeholders',
+                                                      custom_placeholder.id)
+            self.session.query(CustomPlaceholder).filter_by(id=custom_placeholder.id).update(uploaded_images)
+
+    view_kwargs = True
+    decorators = (api.has_permission('is_admin', methods="POST"),)
+    schema = CustomPlaceholderSchema
+    data_layer = {'session': db.session,
+                  'model': CustomPlaceholder,
+                  'methods': {
+                      'query': query,
+                      'before_create_object': before_create_object,
+                      'after_create_object': after_create_object
+                  }}
+
+
+class CustomPlaceholderDetail(ResourceDetail):
+    """
+    custom placeholder detail by id
+    """
+
+    def before_patch(self, args, view_kwargs, data):
+        event_sub_topic = None
+        if view_kwargs.get('event_sub_topic_id'):
+            event_sub_topic = safe_query(self, EventSubTopic, 'id', view_kwargs['event_sub_topic_id'],
+                                         'event_sub_topic_id')
+
+        if event_sub_topic:
+            custom_placeholder = safe_query(self, CustomPlaceholder, 'event_sub_topic_id', event_sub_topic.id,
+                                            'event_sub_topic_id')
+            view_kwargs['id'] = custom_placeholder.id
+
+    def before_get_object(self, view_kwargs):
```

Comment:
You are doing the same thing in before_patch and before_get_object. Please correct me If I'm wrong but since get_object is part of sqlalchemy data layer it uses this same method to get the object for patch request as well so if you are setting up `id` here is get method then you don't need patch one.